src/components/Images/Images.js

```diff
@@ -67,236 +66,99 @@ export const CloudinaryImage = ({
-  const tabletUrlJp2_3x = cld.url(publicIdFilename, {...tabletImageJp2, ...dpr3Setting})
-  const tabletUrlJpeg_3x = cld.url(publicIdFilename, {...tabletImageJpeg, ...dpr3Setting})
-
-  // ------------------------------
-  // Laptop Settings
-  // ------------------------------
-  const laptopImageSettings = {
-    ...baseImageSettings,
-    ...(!!width[2]) && { width: width[2]},
-    ...(!!height[2]) && { height: height[2]},
-  }
-
-  const laptopUrl = cld.url(publicIdFilename, laptopImageSettings)
-
-  const laptopImageWebP = {
-    ...laptopImageSettings,
-    format: 'webp'
-  }
-
-  const laptopImageJp2 = {
-    ...laptopImageSettings,
-    format: 'jp2'
-  }
-
-  const laptopImageJpeg = {
-    ...laptopImageSettings,
-    format: 'jpeg'
-  }
-
-  const laptopUrlWebP = cld.url(publicIdFilename, {...laptopImageWebP, ...dpr1Setting})
-  const laptopUrlJp2 = cld.url(publicIdFilename, {...laptopImageJp2, ...dpr1Setting})
-  const laptopUrlJpeg = cld.url(publicIdFilename, {...laptopImageJpeg, ...dpr1Setting})
-
-  const laptopUrlWebP_2x = cld.url(publicIdFilename, {...laptopImageWebP, ...dpr2Setting})
-  const laptopUrlJp2_2x = cld.url(publicIdFilename, {...laptopImageJp2, ...dpr2Setting})
-  const laptopUrlJpeg_2x = cld.url(publicIdFilename, {...laptopImageJpeg, ...dpr2Setting})
-
-  const laptopUrlWebP_3x = cld.url(publicIdFilename, {...laptopImageWebP, ...dpr3Setting})
-  const laptopUrlJp2_3x = cld.url(publicIdFilename, {...laptopImageJp2, ...dpr3Setting})
-  const laptopUrlJpeg_3x = cld.url(publicIdFilename, {...laptopImageJpeg, ...dpr3Setting})
-
-  // ------------------------------
-  // Desktop Settings
-  // ------------------------------
-  const desktopImageSettings = {
-    ...baseImageSettings,
-    ...(!!width[3]) && { width: width[3]},
-    ...(!!height[3]) && { height: height[3]},
-  }
-
-  const desktopUrl = cld.url(publicIdFilename, desktopImageSettings)
-
-  const desktopImageWebP = {
-    ...desktopImageSettings,
-    format: 'webp'
-  }
-
-  const desktopImageJp2 = {
-    ...desktopImageSettings,
-    format: 'jp2'
-  }
-
-  const desktopImageJpeg = {
-    ...desktopImageSettings,
-    format: 'jpeg'
+  const fileFormats = ['webp', 'jp2', 'jpeg']
+
+  const dprSettings = ['1.0', '2.0', '3.0']
+
+  const mediaBreakpoints = [
+    Media.BREAKPOINTS.DESKTOP_RANGE_START,
+    Media.BREAKPOINTS.LAPTOP_RANGE_START,
+    Media.BREAKPOINTS.TABLET_RANGE_START,
+    Media.BREAKPOINTS.PHONE_RANGE_END,
+  ]
+
+  // We are expecting width/height attribute arrays in the order of
+  // Phone/Tablet/Laptop/Desktop but we have to setup media queries
+  // in the opposite order, so we reverse the arrays here.
+  if (width) {
+    width.reverse()
+  }
+  if (height) {
+    height.reverse()
+  }
+
+  let sourceTags = []
+  const srcSetAttribute = lazyload ? 'data-srcset' : 'srcSet'
+  const srcAttribute = lazyload ? 'data-src' : 'src'
+
+  for (var format = 0; format < fileFormats.length; format++) {
+    for (
```

Comment:
Not usually a fan of nested loops, but I guess we know ahead these will be finite and small correct?

Also, can we create some test coverage for all the conditional logic in here and possibly make this more testable and less procedural by breaking it out into meaningful methods/functions?